packages/samples/headless-react/src/pages/SearchPage.tsx

```diff
@@ -0,0 +1,373 @@
+      configuration: HeadlessEngine.getSampleConfiguration(),
+      reducers: recommendationAppReducers,
+    });
+
+    this.recommendationList = buildRecommendationList(
+      this.recommendationEngine
+    );
+
+    this.tabs = {
+      all: buildTab(this.engine, {
+        initialState: {isActive: true},
+        options: {expression: ''},
+      }),
+      messages: buildTab(this.engine, {
+        options: {expression: '@objecttype==Message'},
+      }),
+      confluence: buildTab(this.engine, {
+        options: {
+          expression:
+            '@connectortype==Confluence2Crawler AND NOT @documenttype==Space',
+        },
+      }),
+    };
+
+    this.searchBox = buildSearchBox(this.engine, {
+      options: {numberOfSuggestions: 8},
+    });
+
+    this.didYouMean = buildDidYouMean(this.engine);
+
+    this.searchStatus = buildSearchStatus(this.engine);
+
+    this.queryError = buildQueryError(this.engine);
+
+    this.querySummary = buildQuerySummary(this.engine);
+
+    this.facetManager = buildFacetManager(this.engine);
+
+    this.geographyFacet = buildCategoryFacet(this.engine, {
+      options: {field: 'geographicalhierarchy'},
+    });
+    this.objectTypeFacet = buildFacet(this.engine, {
+      options: {field: 'objecttype'},
+    });
+
+    this.fileSizeAutomaticNumericFacet = buildNumericFacet(this.engine, {
+      options: {
+        field: 'size',
+        facetId: 'size-3',
+        generateAutomaticRanges: true,
+      },
+    });
+    this.fileSizeManualNumericFacet = buildNumericFacet(this.engine, {
+      options: {
+        field: 'size',
+        facetId: 'size-4',
+        generateAutomaticRanges: false,
+        currentValues: [
+          buildNumericRange({start: 0, end: 5 * KB}),
+          buildNumericRange({start: 5 * KB, end: 5 * MB}),
+          buildNumericRange({start: 5 * MB, end: 5 * GB}),
+        ],
+      },
+    });
+
+    this.sort = buildSort(this.engine, {
+      initialState: {criterion: initialCriterion},
+    });
+
+    this.resultList = buildResultList(this.engine);
+
+    this.resultsPerPage = buildResultsPerPage(this.engine, {
+      initialState: {numberOfResults: resultsPerPageOptions[0]},
+    });
+
+    this.pager = buildPager(this.engine, {options: {numberOfPages: 6}});
+
+    this.history = buildHistory(this.engine);
+
+    this.relevanceInspector = buildRelevanceInspector(this.engine);
+  }
+
+  componentDidMount() {
+    // Search parameters are restored once `SearchParametersManager` is registered,
+    // which in this sample corresponds with when `bindSearchParametersToURI is called.
+    // Beware not to restore search parameters until all components are registered,
+    // otherwise some components such as facets may not work when the page loads.
+    // In this sample, `SearchPage.componentDidMount` happens to be executed after
+    // all components are registered, but depending on your implementation this may
+    // not be your case.
```

Comment:
I changed this comment here to better reflect what's going on with the search parameters and the `componentDidMount` lifecycle method.

It may be useful to include some of this information redundantly in some tutorials or other documentation.

@fbeaudoincoveo

Reply:
Thanks for the heads up @btaillon !

@kaitlynkatz